pkg/rendering/render.go

```diff
@@ -0,0 +1,8 @@
+package rendering
+
+// Engine is an interface representing
+// a generic rendering engine
+type Engine interface {
```

Comment:
Interface should be with consumer: https://github.com/golang/go/wiki/CodeReviewComments#interfaces

Accept interfaces, return structs.

Reply:
There is an interface in the reconciler now, which is a bit useless, since there is only one template engine.